Sources/TuistDependencies/SwiftPackageManager/Utils/SwiftPackageManagerGraphGenerator.swift

```diff
@@ -1,13 +1,18 @@
 import Foundation
+import ProjectDescription
 import TSCBasic
 import TSCUtility
 import TuistCore
 import TuistGraph
+import TuistLoader
```

Comment:
this doesn't adhere to our [architecture](https://docs.tuist.io/contributors/architecture).

We shouldn't use `TuistLoader` in `TuistDependencies` framework. You could either:
a) Return the `ProjectDescription` models from the `DependenciesController.fetch` call, convert them and then create a new function for `DependenciesController` called e.g. `generateDepedenciesGraph` (or something like that) that will take as an input the converted models and serialize + write the graph. 
b) Pass the closure of how to convert the models: `convertProject: (ProjectDescription.Project) -> TuistGraph.Project`

a) would be imho better (feel free to deviate from the suggestion, I have not delved too much into the code)
b) might be simpler but means passing around the closure, which is not the nicest solution of all, IMHO

Reply:
@fortmarek I went for option (a).
It required quite a bit of changes, but I think it is worth it 🙌 
Changes here: https://github.com/tuist/tuist/pull/3099/commits/d959e220b8eeee090336d8eaad6d20aff691766e